app/src/test/java/com/keyboardbuilder/dto/KeyboardCaseTest.java

```diff
@@ -0,0 +1,71 @@
+package com.keyboardbuilder.dto;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class KeyboardCaseTest {
+
+    private KeyboardCase case1 = null;
+
+    @Test
+    void EmptyConstructorNotNull()
+    {
+        givenEmptyCaseConstruct();
+        thenCaseNotNull();
+    }
+
+    @Test
+    void FullConstructor()
+    {
+        givenMaterialAndLayout("Wood", 3);
+        thenMaterialEquals("Wood");
+        thenLayoutEquals(3);
+    }
+
+    @Test
+    void GetterSetterTest()
+    {
+        givenEmptyCaseConstruct();
+        whenMaterialSet("Plastic");
+        whenLayoutSet(2);
+        thenMaterialEquals("Plastic");
+        thenLayoutEquals(2);
```

Comment:
Using Given-When-Then structure to organize tests.
3. Given-when-then test structure.